lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useState, useContext, useEffect, useRef } from 'react';
+import classnames from 'classnames';
+import NxToastContainerContext from './contexts';
+
+import { nxToastPropTypes, NxToastProps } from './types';
+
+const NxToast = (props: NxToastProps) => {
+  const { onClose, children } = props;
+
+  const [isClosing, setIsClosing] = useState(false);
+
+  const toastContainerContextValue = useContext(NxToastContainerContext);
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    ref.current?.querySelector<HTMLElement>('.nx-btn--close')?.focus();
+  }, []);
+
+  const handleClose = () => {
+    if (!isClosing) {
+      setIsClosing(true);
+      toastContainerContextValue?.onToastClosing(ref.current);
+    }
+  };
+
+  const handleAnimationEnd = () => {
+    if (isClosing) {
+      onClose();
+    }
+  };
+
+  const validChild = React.Children.only(children),
+      childrenWithProps = React.cloneElement(validChild, { onClose: handleClose });
+
+  const classes = classnames('nx-toast', {
+    'nx-toast--closing': isClosing
```

Comment:
Like most of our components, we should probably have the ability for the caller to add their own classnames to this component.

Reply:
> Like most of our components, we should probably have the ability for the caller to add their own classnames to this component.

Would you rather it be an explicit prop listed under NxToastProps, or have NxToastProps extend HTMLAttributes<HTMLDivElement> in order to accept a className? Looking through the RSC I've seen it done both ways and unsure which method is best in this case.

Reply:
> Would you rather it be an explicit prop listed under NxToastProps, or have NxToastProps extend HTMLAttributes in order to accept a className? Looking through the RSC I've seen it done both ways and unsure which method is best in this case.

Might as well have it accept all global HTML attributes

Reply:
> Might as well have it accept all global HTML attributes

Implemented here: [5c7a00b](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/5c7a00bd2bf5d6a145ad349efd8921fd5ebd6e14)

Reply:
> Implemented here: [5c7a00b](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/5c7a00bd2bf5d6a145ad349efd8921fd5ebd6e14)

You forgot to apply all of the extra props to the `div`

Reply:
> You forgot to apply all of the extra props to the `div`

added in here: [03a01e5](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/03a01e59e7f824be9508353c7c3477979a8a2ffe)